pages/electronics-library.tsx

```diff
@@ -109,27 +116,139 @@ const Home = ({
   const [activeProduct, setActiveProduct] =
     useState<IElectronicsComponent | null>(null);
 
+  const [searchField, setSearchField] = useState<string>('');
+
+  const [timerId, setTimerId] = useState<ReturnType<typeof setTimeout>>();
+
+  const [searchCategoryTags, setSearchCategoryTags] = useState<CategoryTags[]>([
+    CategoryTags.ShowAll,
+  ]);
+  const onSearchChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const searchValue = e.currentTarget.value;
+    clearTimeout(timerId);
+    setTimerId(
+      setTimeout(() => {
+        setSearchField(searchValue);
+      }, 500)
+    );
+  };
+  const onTagSelect = (e: React.FormEvent<HTMLInputElement>) => {
+    setSearchCategoryTags([e.currentTarget.value as CategoryTags]);
+  };
+
+  const [filteredComponents, setFilteredComponents] =
+    useState<IElectronicsComponent[]>(electronicComponents);
+  const filterComponents = () => {
+    let componentList: IElectronicsComponent[] = electronicComponents;
+    if (searchCategoryTags[0] !== CategoryTags.ShowAll) {
+      componentList = componentList.filter((component) => {
+        return component.category
+          ? component.category.toLowerCase() ===
+              searchCategoryTags[0].toLowerCase()
+          : null;
+      });
+    }
+
+    if (searchField) {
+      const searcher = new Searcher(componentList, {
+        keySelector: (obj: IElectronicsComponent) =>
+          [
+            obj.category,
+            obj.name,
+            obj.shortDescription,
+            obj.description,
+            obj.tags?.join(' '),
+            obj.shownTo,
+            obj.leadTime,
+          ].filter((item: string | undefined): item is string => !!item),
```

Comment:
Okay this was a fun Typescript error. Not sure why it got ignored in -dev version...

Removing the optional parameter ("?") in the interface fixed it for me. This should be okay since in getStaticProps empty fields are removed with removeEmpty and then a placeholder is used to fill in any missing fields.
![image](https://user-images.githubusercontent.com/129784422/231565643-abab68b2-2e23-40b0-bebc-f9551894e78e.png)

![image](https://user-images.githubusercontent.com/129784422/231568681-c30bcccf-008e-47aa-b821-3b1f3a8750fc.png)

Reply:
This was the most mysterious thing that's happened to me this week.